lib/streamlit/elements/image.py

```diff
@@ -301,8 +302,34 @@ def image_to_url(
     If `image` is already a URL, return it unmodified.
     Otherwise, add the image to the MediaFileManager and return the URL.
     """
+
+    image_data: bytes
+
+    # Strings
+    if isinstance(image, str):
+        # If it's a url, return it directly.
+        try:
+            p = urlparse(image)
+            if p.scheme:
+                return image
+        except UnicodeDecodeError:
+            pass
+
+        # Otherwise, try to open it as a file.
+        try:
+            with open(image, "rb") as f:
+                image_data = f.read()
+        except:
+            # When we aren't able to open the image file, we still pass the path to
+            # the MediaFileManager - its storage backend may have access to files
+            # that Streamlit does not.
+            mimetype, _ = mimetypes.guess_type(image)
+            if mimetype is None:
+                mimetype = "application/octet-stream"
+            return media_file_manager.add(image, mimetype, image_id)
```

Comment:
Does this imply that certain image operations won't be supported if Streamlit doesn't have access to the file but the storage backend does?

I don't think we're far along enough in the interface-ization of `MediaFileManager` to need to document this anywhere, but we'll probably want to mention it somewhere once we're a bit further along

Reply:
Yeah - I've given Edward Ma a heads up on this, so he's aware. I'll also include docs about this in a future PR.